app/javascript/components/ComboCommunesSearch.jsx

```diff
@@ -1,9 +1,28 @@
 import React from 'react';
 import { QueryClientProvider } from 'react-query';
+import { matchSorter } from 'match-sorter';
 
 import ComboSearch from './ComboSearch';
 import { queryClient } from './shared/queryClient';
 
+function expandResultsWithMultiplePostalCodes(term, results) {
+  const expandedResults = results.flatMap((result) =>
+    result.codesPostaux.map((codePostal) => ({
+      ...result,
+      codesPostaux: [codePostal]
+    }))
+  );
+  const limit = term.length > 5 ? 10 : 5;
```

Comment:
```suggestion

  // Some very large towns (like Paris) have A LOT of associated postal codes.
  // As we generated one result per postal code, we now have a lot of results
  // for the same town.
  //
  // If we have a large number of results, ????
  const limit = term.length > 5 ? 10 : 5;
```

Je ne comprends pas exactement ce que fait ce code ("on ajoute une recherche locale pardessus la liste augmentée" ?) ; du coup je te laisse compléter le "?????" :)